File: VB/Dashboard_ConditionalFormatting_Grid/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin

Namespace Grid_IconRangeCondition
    Partial Public Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
            Dim dashboard As New Dashboard()
            dashboard.LoadFromXml("..\..\Data\Dashboard.xml")
            dashboardViewer1.Dashboard = dashboard
            Dim grid As GridDashboardItem = CType(dashboard.Items("gridDashboardItem1"), GridDashboardItem)
            Dim extendedPrice As GridMeasureColumn = CType(grid.Columns(1), GridMeasureColumn)

            Dim barRule As New GridItemFormatRule(extendedPrice)
            Dim barCondition As New FormatConditionBar()
            barCondition.StyleSettings = New BarStyleSettings(FormatConditionAppearanceType.PaleBlue)
            barCondition.NegativeStyleSettings = New BarStyleSettings(FormatConditionAppearanceType.PaleRed)
            barCondition.BarOptions.AllowNegativeAxis = checkBox1.Checked
            barCondition.BarOptions.DrawAxis = True
            barRule.Condition = barCondition

            grid.FormatRules.AddRange(barRule)
        End Sub

        Private Sub checkBox1_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) _
            Handles checkBox1.CheckedChanged
            UpdateBarOptions()
        End Sub

        Private Sub checkBox2_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) _
            Handles checkBox2.CheckedChanged
            UpdateBarOptions()
        End Sub

        Private Sub checkBox3_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) _
            Handles checkBox3.CheckedChanged
            UpdateBarOptions()
        End Sub

        Private Sub UpdateBarOptions()
            Dim grid As GridDashboardItem = CType(dashboardViewer1.Dashboard.Items(0), GridDashboardItem)
            Dim barOptions As FormatConditionBarOptions =
                CType((grid.FormatRules(0)).Condition, FormatConditionBar).BarOptions
            barOptions.AllowNegativeAxis = checkBox1.Checked
            barOptions.DrawAxis = checkBox2.Checked
            barOptions.ShowBarOnly = checkBox3.Checked
        End Sub
    End Class
End Namespace

File: CS/Dashboard_ConditionalFormatting_Grid/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;

namespace Grid_IconRangeCondition {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm {
        public Form1() {
            InitializeComponent();
            Dashboard dashboard = new Dashboard(); dashboard.LoadFromXml(@"..\..\Data\Dashboard.xml");
            dashboardViewer1.Dashboard = dashboard;
            GridDashboardItem grid = (GridDashboardItem)dashboard.Items["gridDashboardItem1"];
            GridMeasureColumn extendedPrice = (GridMeasureColumn)grid.Columns[1];

            GridItemFormatRule barRule = new GridItemFormatRule(extendedPrice);
            FormatConditionBar barCondition = new FormatConditionBar();
            barCondition.StyleSettings = new BarStyleSettings(FormatConditionAppearanceType.PaleBlue);
            barCondition.NegativeStyleSettings = new BarStyleSettings(FormatConditionAppearanceType.PaleRed);
            barCondition.BarOptions.AllowNegativeAxis = checkBox1.Checked;
            barCondition.BarOptions.DrawAxis = true;
            barRule.Condition = barCondition;

            grid.FormatRules.AddRange(barRule);
        }

        private void checkBox1_CheckedChanged(object sender, System.EventArgs e) {
            UpdateBarOptions();
        }

        private void checkBox2_CheckedChanged(object sender, System.EventArgs e) {
            UpdateBarOptions();
        }

        private void checkBox3_CheckedChanged(object sender, System.EventArgs e) {
            UpdateBarOptions();
        }

        private void UpdateBarOptions() {
            GridDashboardItem grid = (GridDashboardItem)dashboardViewer1.Dashboard.Items[0];
            FormatConditionBarOptions barOptions = ((FormatConditionBar)(grid.FormatRules[0]).Condition).BarOptions;
            barOptions.AllowNegativeAxis = checkBox1.Checked;
            barOptions.DrawAxis = checkBox2.Checked;
            barOptions.ShowBarOnly = checkBox3.Checked;
        }
    }
}
